src/SFA.DAS.ApplyService.Data/ApplyRepository.cs

```diff
@@ -796,5 +980,95 @@ public async Task<string> GetNextRoatpApplicationReference()
                 return $"APR{nextInSequence}";
             }
         }
+
+        public async Task<bool> RecordFinancialGrade(Guid applicationId, FinancialReviewDetails financialReviewDetails, string financialReviewStatus)
+        {
+            using (var connection = new SqlConnection(_config.SqlConnectionString))
+            {
+                try
+                {
+                    var result = await connection.QueryAsync<ApplyData>(@"SELECT ApplyData FROM Apply WHERE ApplicationId = @applicationId",
+                                                                            new { applicationId });
+                    var applyData = result.FirstOrDefault();
+                    if (applyData == null)
+                    {
+                        _logger.LogError($"Unable to record financial grade for application {applicationId} - no apply data found");
+                        return await Task.FromResult(false);
+                    }
+                    applyData.FinancialReviewDetails = financialReviewDetails;
+                    
+                    var recordsAffected = await connection.ExecuteAsync(@"UPDATE Apply 
+                                                                        SET ApplyData = @applyData,
+                                                                        FinancialGrade = @grade
+                                                                        WHERE ApplicationId = @applicationId",
+                        new
+                        {
+                            applicationId,
+                            applyData,
+                            grade = financialReviewDetails.SelectedGrade
+                        });
+                    if (recordsAffected > 0)
+                    {
+                        await connection.ExecuteAsync("UPDATE Apply SET FinancialReviewStatus = @financialReviewStatus WHERE ApplicationId = @applicationId",
```

Comment:
Does this need to be done as a separate call, especially since you know that ApplyData already exists and thus will be recordsAffected will equal 1? If so can we have a comment in the code

Reply:
Good call, compressed into a single update query